### PR TITLE
bpo-44282: fix occasional test_incremental_editing failures on buildbots

### PR DESCRIPTION
**Pull request by taleinat**
https://bugs.python.org/issue44282

### CODE REVIEW

File: Lib/idlelib/idle_test/test_colorizer.py [hunk 1]
```diff
@@ -551,7 +551,7 @@ def test_long_multiline_string(self):
             ''')
         self._assert_highlighting(source, {'STRING': [('1.0', '5.4')]})
 
-    @run_in_tk_mainloop
+    @run_in_tk_mainloop(delay=50)
```
Comment (vstinner) [inline Lib/idlelib/idle_test/test_colorizer.py:554]: What is this magic number? A number of seconds?

I suggest you to use test.support.SHORT_TIMEOUT or test.support.LONG_TIMEOUT: https://docs.python.org/dev/library/test.html#test.support.SHORT_TIMEOUT

Comment (taleinat) [reply Lib/idlelib/idle_test/test_colorizer.py:554]: > What is this magic number? A number of seconds?

Milliseconds. It's explained in the decorator's doc-string.

> I suggest you to use test.support.SHORT_TIMEOUT or test.support.LONG_TIMEOUT

I wasn't aware of these, good to know! I don't think they're quite what's needed here though.

Comment (taleinat) [reply Lib/idlelib/idle_test/test_colorizer.py:554]: > > What is this magic number? A number of seconds?
> 
> Milliseconds. It's explained in the decorator's doc-string.

Would you prefer that this was a `datetime.timedelta` instance?

**Review by vstinner — APPROVED** [review]
LGTM. Well, I don't understand how Tkinter handles time, so let's see if 50 ms is enough for everyone :-p

File: Lib/idlelib/idle_test/test_colorizer.py (hunk 1 above)
Comment (vstinner) [reply Lib/idlelib/idle_test/test_colorizer.py:554]: No, it's fine.

**Review by terryjreedy — APPROVED** [review]
Exactly what I meant as an alternative to skipping.  While the latter would be okay for this test, we may have other tests where a much smaller delay, even 2 or 3 is sufficient.

**Comment by vstinner** [comment]
I merge the issue right now since it broke buildbots and I'm trying to recover the green color of buildbots!

**Comment by terryjreedy** [comment]
A note to myself: Backport the revised tkinter helper file to 3.9 so we can use it with future tests that get backported to 3.9.

**Comment by taleinat** [comment]
> A note to myself: Backport the revised tkinter helper file to 3.9 so we can use it with future tests that get backported to 3.9.

Note that it was added as part of the PR which added the shell sidebar. We'd likely want to backport that to 3.9 first.

**Pull request merged**